hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -219,17 +226,31 @@ class MergeOnReadSnapshotRelation(val sqlContext: SQLContext,
           } else {
             Option.empty
           }
-
           val logPaths = fileSlice.getLogFiles.sorted(HoodieLogFile.getLogFileComparator).iterator().asScala
             .map(logFile => MergeOnReadSnapshotRelation.getFilePath(logFile.getPath)).toList
           val logPathsOptional = if (logPaths.isEmpty) Option.empty else Option(logPaths)
-          HoodieMergeOnReadFileSplit(partitionedFile, logPathsOptional, queryInstant, metaClient.getBasePath,
-            maxCompactionMemoryInBytes, mergeType)
+          (fileSlice.getBaseFile, HoodieMergeOnReadFileSplit(partitionedFile, logPathsOptional, queryInstant, metaClient.getBasePath,
+            maxCompactionMemoryInBytes, mergeType))
         }).toList
-        fileSplits
+        groupFilesWithBucket(fileSplits)
       }
     }
   }
+
+  private def groupFilesWithBucket(
```

Comment:
can you explain why this needs to happen? At a high level, I would expect bucketing not alter how we group files? we have 1 file group per bucket right?

Reply:
This is a read optimizer. I revert this changes and will create another pr to cover the query optimizer part.